consensus/istanbul/core/roundstate.go

```diff
@@ -98,6 +102,9 @@ type roundStateImpl struct {
 	// to be persisted.
 	proposalVerificationStatus map[common.Hash]error
 
+	// Cache for StateProcessResult in this sequence.
+	stateProcessResults map[common.Hash]*StateProcessResult
```

Comment:
This cache implementation does not have any cache invalidation and will slowly leak memory. IIUC the cache only needs to hold the most recent value. If this is true, I recommend changing this to be a single result object, along with it's block hash. (i.e. have it be a singleton cache) If more values are needed, I recommend building a simple circular buffer implementation that stores the results for up to `k` blocks, indexed by `blockNumber % k`.

Reply:
Not sure if that's really needed. on `StartNewSequence` clear this map. So you only cache values for the current sequence. You could argue that if there are many rounds, and each round implies a new block, then this could get bigger. But not sure if we want to worry about that (or at least we would have other things to worry if that happened)

Reply:
Oh, great! I missed didn't notice that when I was looking for cache invalidation logic. 👍